dev/requirements.txt

```diff
@@ -6,3 +6,13 @@ pydata_sphinx_theme
 ipython
 nbsphinx
 numpydoc
+
+# dependencies in pandas-on-spark.
```

Comment:
@xinrong-databricks 
Do you intend to include dev dependencies in this PR?

Reply:
Yeah, this file is only used when setting up a dev env for now.

Reply:
Yes, this PR includes dev dependencies.